lib/temporal/workflow/context.rb

```diff
@@ -171,12 +176,32 @@ def cancel_timer(timer_id)
       def complete(result = nil)
         command = Command::CompleteWorkflow.new(result: result)
         schedule_command(command)
+        completed!
       end
 
       # TODO: check if workflow can be failed
       def fail(exception)
         command = Command::FailWorkflow.new(exception: exception)
         schedule_command(command)
+        completed!
+      end
+
+      def continue_as_new(workflow_class, *input, **args)
```

Comment:
I'm assuming most of the times you'd want to "continue as new" the same workflow, right? In which case let's remove `workflow_class` for the arguments and just use `self.class` instead. This can be overridden via passing everything via `options` if needed

Reply:
I'm not able to skip the argument because the context does not appear to know which workflow class it's for. `self.class` is `Temporal::Workflow::Context` not the workflow class whose `execute` we are running.

Reply:
Should we fix the construction of the context so that it stores the workflow class?

Reply:
I prefer it as it is personally as that's the upstream API and matches other SDKs.